extensions/guacamole-auth-quickconnect/src/main/java/org/apache/guacamole/auth/quickconnect/QuickConnectAuthenticationProvider.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.guacamole.auth.quickconnect;
+
+import java.util.Collections;
+import java.util.Map;
+import org.apache.guacamole.GuacamoleException;
+import org.apache.guacamole.net.auth.AuthenticatedUser;
+import org.apache.guacamole.net.auth.Credentials;
+import org.apache.guacamole.net.auth.credentials.CredentialsInfo;
+import org.apache.guacamole.net.auth.credentials.GuacamoleInvalidCredentialsException;
+import org.apache.guacamole.net.auth.simple.SimpleAuthenticationProvider;
+import org.apache.guacamole.net.auth.UserContext;
+import org.apache.guacamole.protocol.GuacamoleConfiguration;
+
+/**
+ * Class providing the necessary hooks into the Guacamole Client authentication
+ * process so that the QuickConnect functionality can be initialized and be used
+ * throughout the web client.
+ */
+public class QuickConnectAuthenticationProvider extends SimpleAuthenticationProvider {
```

Comment:
As you have your own underlying `UserContext` implementation (a good thing), you should not be extending `SimpleAuthenticationProvider` here.

Reply:
...even though doing so greatly reduces the amount of code I have to write to overwrite methods that are doing exactly the same thing as `SimpleAuthenticationProvider` - like `shutdown()` and `decorate()` and `redecorate()` and probably a couple of others that I would basically just be copying/pasting code out of `SimpleAuthenticationProvider` to duplicate that functionality?

Reply:
I get where you're coming from, and reducing duplication is good, but what you need is an actual base implementation of `AuthenticationProvider` which guarantees the default behavior you're looking for. `SimpleAuthenticationProvider` is not guaranteed or intended to provide this, and is only defined with respect to `getAuthorizedConfigurations()`.

A base, empty implementation of `AuthenticationProvider` would probably be a good thing (similar to `AbstractUser` and others). If you're looking to avoid duplication, perhaps we should:

* Write an `AbstractAuthenticationProvider` (perhaps also `AbstractUserContext` if it helps)
* Refactor `SimpleAuthenticationProvider` to build off that if appropriate.
* Avoid using `SimpleAuthenticationProvider` here

Reply:
Ah, okay, that makes sense.  I guess I'll just switch it over to `implements AuthenticationProvider` for now - I think it would probably make sense at some point in the future to create the `AbstractAuthenticationProvider` and `AbstractUserContext` implementations, but it seems like that might be more appropriate in a separate change.

Reply:
Implemented.